src/Test/QuickCheck/Gen.purs

```diff
@@ -160,9 +160,13 @@ runGen = runState
 evalGen :: forall a. Gen a -> GenState -> a
 evalGen = evalState
 
--- | Sample a random generator
-sample :: forall r a. Seed -> Size -> Gen a -> Array a
-sample seed sz g = evalGen (vectorOf sz g) { newSeed: seed, size: sz }
+-- | Sample a random generator, using the given seed and producing a sample
+-- | containing the given number of values.
+sample :: forall r a. Seed -> Int -> Gen a -> Array a
+sample seed n g = evalGen sampleGen { newSeed: seed, size: 0 }
+  where
+  sampleGen = traverse (flip resize g) sizes
+  sizes = range 0 (n - 1) <#> (*2)
```

Comment:
I'm not sure - a linear increasing rate doesn't sound *too* fast to me (admittedly this is based on nothing other than a gut feeling). Also, using 20 as a cap seems a bit arbitrary (ha) and potentially surprising in some situations. Haskell's might only be capped at 20 because `sample` is hardcoded to only produce a certain number of results (I think you're right that it's 11). I'll try to work out what sizes get chosen when you use `quickCheck` in Haskell, perhaps that will help us agree on a solution?

Perhaps it would also help to discuss what we think `sample` should be used for. Should we eventually change `quickCheck` to use sample internally, for example? Or should it only be for testing out `Arbitrary` instances that you've written? Or should it be regarded as a slightly nicer API than `evalGen` to be used in pretty much any situation where you want randomly generated value? Maybe it would make sense to allow the caller of `sample` to supply the function `f` like in your example?